sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/BufferHolder.java

```diff
@@ -55,6 +58,15 @@ public BufferHolder(UnsafeRow row, int initialSize) {
     this.buffer = new byte[fixedSize + initialSize];
     this.row = row;
     this.row.pointTo(buffer, buffer.length);
+    this.array = null;
+  }
+
+  public BufferHolder(UnsafeArrayData array, long numElements) {
```

Comment:
I do not understand what `this` means.

Do you want not to allocate `BufferHolder` for this case. If so, it requires to refactor `UnsafeArrayWrite` since the current `UnsafeArrayWrite` assumes `BufferHolder`.

Or , do you want to eliminate `numElements`?

Or, do you want to allocate `buffer` here? I did not take this approach since it introduces a duplication of the same logic here and in `UnsafeArrayWrite`.